pyomo/contrib/parmest/parmest.py

```diff
@@ -613,18 +1181,234 @@ def _Q_opt(
+                    elif (
+                        cov_method == CovarianceMethodLib.finite_difference
+                        or cov_method
+                        == CovarianceMethodLib.automatic_differentiation_kaug
+                    ):
+                        cov = compute_covariance_matrix(
+                            self.exp_list,
+                            method,
+                            theta_vals=self.estimated_theta,
+                            solver=solver,
+                            step=step,
+                            tee=self.tee,
+                        )
+                    else:
+                        raise NotImplementedError(
+                            'Only "finite_difference", "reduced_hessian", and '
+                            '"automatic_differentiation_kaug" methods are supported.'
+                        )
+                else:
+                    raise ValueError(
+                        "One or more values of the measurement errors have "
+                        "not been supplied."
+                    )
+            else:
+                raise AttributeError(
+                    'Experiment model does not have suffix "measurement_error".'
+                )
+        elif self.obj_function == ObjectiveLib.SSE_weighted:
+            # check if the user defined the 'measurement_error' attribute
+            if hasattr(model, "measurement_error"):
+                meas_error = [
+                    model.measurement_error[y_hat]
+                    for y_hat, y in model.experiment_outputs.items()
+                ]
+
+                # check if the user supplied the values for the measurement errors
+                if all(item is not None for item in meas_error):
+                    if (
+                        cov_method == CovarianceMethodLib.finite_difference
+                        or cov_method
+                        == CovarianceMethodLib.automatic_differentiation_kaug
+                    ):
+                        cov = compute_covariance_matrix(
+                            self.exp_list,
+                            method,
+                            theta_vals=self.estimated_theta,
+                            step=step,
+                            solver=solver,
+                            tee=self.tee,
+                        )
+                    elif cov_method == CovarianceMethodLib.reduced_hessian:
+                        cov = self.inv_red_hes
+                        cov = pd.DataFrame(
+                            cov,
+                            index=self.estimated_theta.keys(),
+                            columns=self.estimated_theta.keys(),
+                        )
+                    else:
+                        raise NotImplementedError(
+                            'Only "finite_difference", "reduced_hessian", and '
+                            '"automatic_differentiation_kaug" methods are supported.'
+                        )
```

Comment:
This is unreachable code. You already check in lines 1274-1280 whether or not a supported method was supplied, and it raises an error if not. So this is redundant.